t/api/04-jobs.t

```diff
@@ -979,6 +979,7 @@ subtest 'handle settings when posting job' => sub {
 subtest 'do not re-generate settings when cloning job' => sub {
     my $job_settings = $jobs->search({test => 'autoupgrade'})->first->settings_hash;
     clone_job_apply_settings([qw(BUILD_SDK= ISO_MAXSIZE=)], 0, $job_settings, {});
+    $job_settings->{is_clone_job} = 1;
```

Comment:
Why wasn't this set here before? On its own the test change looks wrong because we never checked it

Reply:
I moved the place where we set `is_clone_job` from `clone_job_apply_settings` to a different function¹ which is not executed as part of this test. So we need to set it manually here. The tests in `t/35-script_clone_job.t` still check whether `is_clone_job` is present without faking it. So this doesn't make our testing worse.

---

¹ From `clone_job_apply_settings` to `post_jobs`. This was necessary because the parameter is not an actual job setting and therefore should not get a `:`-suffix.